mullvad-relay-selector/src/error.rs

```diff
@@ -0,0 +1,62 @@
+//! Definition of relay selector errors
+#![allow(dead_code)]
+
+use mullvad_types::{relay_constraints::MissingCustomBridgeSettings, relay_list::Relay};
+
+use crate::WireguardConfig;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("Failed to open relay cache file")]
+    OpenRelayCache(#[source] std::io::Error),
+
+    #[error("Failed to write relay cache file to disk")]
+    WriteRelayCache(#[source] std::io::Error),
+
+    #[error("No relays matching current constraints")]
+    NoRelay,
+
+    #[error("No bridges matching current constraints")]
+    NoBridge,
+
+    #[error("No obfuscators matching current constraints")]
+    NoObfuscator,
+
+    #[error("No endpoint could be constructed for relay {0:?}")]
+    NoEndpoint(Box<EndpointError>),
+
+    #[error("Failure in serialization of the relay list")]
+    Serialize(#[from] serde_json::Error),
+
+    #[error("Downloader already shut down")]
+    DownloaderShutDown,
+
+    #[error("Invalid bridge settings")]
+    InvalidBridgeSettings(#[from] MissingCustomBridgeSettings),
+}
+
+/// Special type which only shows up in [`Error`]. This error variant signals that no valid
+/// endpoint could be constructed from the selected relay. See [`detailer`] for more info.
+///
+/// [`detailer`]: mullvad_relay_selector::relay_selector::detailer.rs
+#[derive(Debug)]
```

Comment:
The doc-link to `detailer` doesn't work for me.

Reply:
This works for me, but only if the `detailer` module is made `pub`. Not sure if you can even doc-link to non-public items.
```suggestion
/// Special type which only shows up in [`Error`]. This error variant signals that no valid
/// endpoint could be constructed from the selected relay. See [`detailer`][d] for more info.
///
/// [d]: crate::relay_selector::detailer
#[derive(Debug)]
```